e2e_deployment_files/nestededge_middleLayer_messaging_amqp.json

```diff
@@ -0,0 +1,123 @@
+                "systemModules": {
+                    "edgeAgent": {
+                        "type": "docker",
+                        "settings": {
+                            "image": "$upstream:443/microsoft/azureiotedge-agent:<Build.BuildNumber>-linux-<Architecture>",
+                            "createOptions": "{\"HostConfig\":{\"LogConfig\":{\"Type\":\"json-file\",\"Config\":{\"max-size\":\"4m\",\"max-file\":\"7\",\"compress\":\"true\"}}}}"
+                        },
+                        "env": {
+                            "UpstreamProtocol": {
+                                "value": "amqp"
+                            }
+                        }
+                    },
+                    "edgeHub": {
+                        "type": "docker",
+                        "settings": {
+                            "image": "$upstream:443/microsoft/azureiotedge-hub:<Build.BuildNumber>-linux-<Architecture>",
+                            "createOptions": "{\"HostConfig\": {\"LogConfig\":{\"Type\":\"json-file\",\"Config\":{\"max-size\":\"4m\",\"max-file\":\"7\",\"compress\":\"true\"}}, \"PortBindings\": {\"8883/tcp\": [{\"HostPort\": \"8883\"}],\"5671/tcp\": [{\"HostPort\": \"5671\"}]}}}"
+                        },
+                        "env": {
+                            "experimentalFeatures__enabled": {
+                                "value": "true"
+                            },
+                            "DeviceScopeCacheRefreshDelaySecs": {
+                                "value": 0
+                            },
+                            "RuntimeLogLevel": {
+                                "value": "debug"
+                            },
+                            "SslProtocols": {
+                                "value": "tls1.2"
+                            }
+                        },
+                        "status": "running",
+                        "restartPolicy": "always"
+                    }
+                },
+                "modules": {
+                    "IoTEdgeAPIProxy": {
+                        "version": "1.0",
+                        "type": "docker",
+                        "status": "running",
+                        "restartPolicy": "always",
+                        "startupOrder": 3,
+                        "settings": {
+                            "image": "$upstream:443/microsoft/azureiotedge-api-proxy:<Build.BuildNumber>-linux-<Architecture>",
+                            "createOptions": "{\"HostConfig\": {\"LogConfig\":{\"Type\":\"json-file\",\"Config\":{\"max-size\":\"4m\",\"max-file\":\"7\",\"compress\":\"true\"}}, \"PortBindings\": {\"443\/tcp\": [{\"HostPort\": \"443\"}]}}}"
+                        },
+                        "env": {
+                            "NGINX_DEFAULT_PORT": {
+                                "value": "443"
+                            }
+                        }
+                    },
+                    "relayer1": {
```

Comment:
How does the relayer fit into these nested edge AMQP test scenarios compared to the message initiators & receivers for the MQTT scenario?

Reply:
There is an E2E test that sends messages from L3->L4->upstream (I think, didn't make it)